src/main/java/pl/myCompany/account/AccountController.java

```diff
@@ -11,58 +15,93 @@
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
+@Slf4j
 public class AccountController {
 
   private AccountService accountService;
 
   @Autowired
   public AccountController(AccountService accountService) {
     this.accountService = accountService;
-  } //po usunieciu AccountController - nie chodzi, NullPointerExpception dla metody post bo brak adnotacji do serwisu
+  }
 
   @GetMapping("/accounts")
-  public Iterable<Account> getAllAccounts() {
-    return accountService.getAccounts(); //
+  public ResponseEntity<List<Account>> getAllAccounts() {
+    return ResponseEntity.ok(accountService.getAccounts());
   }
 
   @GetMapping("/account/{id}")
-  public Account getSpecificAccount(@PathVariable Long id) {
-    return accountService.getAccount(id);
+  public ResponseEntity<Optional<Account>> getSpecificAccount(@PathVariable Long id) {
+    ResponseEntity<Optional<Account>> response = checkIfAccountExists(id);
+    if (response.equals(ResponseEntity.notFound().build())) {
```

Comment:
I will check if optional which is returned by accountService.getAccount(id) is empty and then return proper http status.